assets/app/features/upgrade/controllers/upgrade-nodes.controller.js

```diff
@@ -100,6 +158,33 @@
             vm.nodesUpgrade.totalNodes = response.data.upgraded_nodes + response.data.remaining_nodes;
             vm.nodesUpgrade.currentNodes = response.data.current_nodes;
             vm.nodesUpgrade.currentAction = response.data.current_node_action;
+            vm.nodesUpgrade.computeNodesPostponedFlag = response.data.compute_nodes_postponed;
+
+            // check for the case of postpone upgrade of compute nodes, after controller nodes were
+            // upgraded, step status would still be 'running' while substep status would be 'finished'
+            var subStep = response.data.current_substep;
+            var subStepStatus = response.data.current_substep_status;
+            var nodesSelected = response.data.nodes_selected_for_upgrade
+            if (subStep === 'controller_nodes' && subStepStatus === 'finished' && nodesSelected === 'controllers') {
+                vm.nodesUpgrade.running = false;
+                // show partially upgrade message
+                vm.nodesUpgrade.upgradeControllersDone = true;
+                upgradeStepsFactory.setUpgradeStep(2);
```

Comment:
Please create constants instead of using literals numbers.

Reply:
will do